### PR TITLE
Thchang/improve help drawer

### PR DESCRIPTION
**Pull request by TienHao**
#795 : changing help drawer pictures in different themes (dark theme picture not added yet)
#679 : automatically showing help drawers on left or right depends on widgets location

### CODE REVIEW

File: src/components/Dialogs/DraggableDialog/DraggableDialogComponent.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class DraggableDialogComponent extends React.Component<ResizableDialogCom
     componentDidUpdate() {
         const header = this.dd.getElementsByClassName("bp3-dialog-header");
         if (this.props.helpType && header.length > 0 && this.dd.getElementsByClassName("help-button").length === 0) {
-            const helpButton = <Button icon="help" minimal={true} onClick={() => this.props.appStore.helpStore.showHelpDrawer(this.props.helpType)}/>;
+            const helpButton = <Button icon="help" minimal={true} onClick={(ev: React.MouseEvent<HTMLButtonElement>) => this.onClickHelpButton(ev)}/>;
```
Comment (veggiesaurus) [inline src/components/Dialogs/DraggableDialog/DraggableDialogComponent.tsx]: can't you just use `onClick={this.onClickHelpButton}` after turning `onClickHelpButton` into an arrow-function? Anonymous functions should be avoided if possible in react, as they always trigger re-rendering

File: src/components/FloatingWidget/FloatingWidgetComponent.tsx [hunk 2]
```diff
@@ -133,7 +137,7 @@ export class FloatingWidgetComponent extends React.Component<FloatingWidgetCompo
                     </div>
                     }
                     {widgetConfig.helpType &&
-                    <div className={buttonClass} onClick={() => appStore.helpStore.showHelpDrawer(widgetConfig.helpType)}>
+                    <div className={buttonClass} onClick={(ev) => this.onClickHelpButton(ev)}>
```
Comment (veggiesaurus) [inline src/components/FloatingWidget/FloatingWidgetComponent.tsx]: same suggestion here

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

**Comment by veggiesaurus** [comment]
@kswang1029 did we resolve to automatically pop up the help drawer on the opposite side to the widget? or let the user decide?

**Comment by kswang1029** [comment]
I think we should try auto.

**Comment by TienHao** [comment]
> @kswang1029 did we resolve to automatically pop up the help drawer on the opposite side to the widget? or let the user decide?

I have implemented automatically left/right popping up by the horizontal location of the help button in widgets and dialogues.

**Comment by veggiesaurus** [comment]
@TienHao what is the logic behind which location it chooses? I'd have thought it would be related to whether the center of the widget is to the left or right of the midpoint, but that's not happening:

image

I see it depends on where you actually _click_ the button, rather than the position of the button itself. I don't think that makes sense. But it's also not switching at the midpoint, so I'm confused as to what the intended logic is.

**Comment by TienHao** [comment]
> I see it depends on where you actually _click_ the button, rather than the position of the button itself. I don't think that makes sense. But it's also not switching at the midpoint, so I'm confused as to what the intended logic is.

@veggiesaurus My previous thought is to use the position of button to decide the help drawer location without blocking the help button.(that did make sense to me at that point. :grin: I think yours better. ) Since the width of drawer is 33%, if the position of help button is over 65% width of browser, the drawer will pop up on left, and less then 65% will pop up on right. 

I have changed the logic to the center of the widget now.

**Comment by veggiesaurus** [comment]
I think we should also consider not applying the dimming overlay, so that you can still see the widget properly

**Review by veggiesaurus — APPROVED** [review]
this looks good. Can you also make a PR for `release/1.3`

**Pull request merged**